CMakeLists.txt

```diff
@@ -53,11 +53,16 @@ add_subdirectory("log4tango")
 add_subdirectory("cppapi")
 
 if(BUILD_TESTING)
-  if(BUILD_SHARED_LIBS)
+  find_program(DOCKER_BINARY docker)
+
+  if(NOT DOCKER_BINARY)
+      message(WARNING "Not building the tests, because docker is not present.")
+      set(BUILD_TESTING OFF)
```

Comment:
Will this handle following scenario without the need to recompile the whole library (server_objects + client_objects)?:
1. `cmake ..`
2. `make`
3. install docker
4. `ctest`

I would just print the warning that the test won't run without docker, without disabling any flags. Lack of docker during configuration time does not prevent us from compiling the tests and device servers. And has nothing to do with presence of docker later, when one may want to run the tests.

Also, we may want to compile individual tests or device server for e.g. debugging or demo purposes and just start the database manually on the host.

If failures with obscure errors are your concern, I suggest to check for docker presence from the test runner script and then print a message and fail the test. What do you think?

Reply:
@mliszcz Thanks for the input. I'll keep the warning but still compile the tests. And I'll fail test execution early if docker is missing.

Reply:
@mliszcz Done.